corehq/apps/reports/tests/test_application_status_report.py

```diff
@@ -0,0 +1,465 @@
+from datetime import datetime, timedelta
+from unittest.mock import patch, MagicMock
+
+from django.test import TestCase, RequestFactory
+
+from corehq.apps.app_manager.models import BuildProfile
+from corehq.apps.domain.models import Domain
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.es.tests.utils import es_test
+from corehq.apps.es.users import user_adapter
+from corehq.apps.locations.models import SQLLocation, LocationType
+from corehq.apps.reports.standard.deployments import ApplicationStatusReport
+from corehq.apps.users.models import CommCareUser, WebUser
+from corehq.util.test_utils import disable_quickcache, flag_enabled
+from corehq.apps.users.dbaccessors import delete_all_users
+from dimagi.utils.parsing import json_format_datetime
+
+
+class MockApplication(object):
+    def __init__(self, _id, domain, name, build_profiles={}):
+        self.domain = domain
+        self.name = name
+        self._id = _id
+        self.build_profiles = build_profiles
+
+    @property
+    def get_id(self):
+        return self._id
+
+
+class TestApplicationStatusReportData(TestCase):
```

Comment:
Why does this subclass `TestCase`? It appears that this class only exists for the `get_user_data` function which is called by tests in other classes. Consider moving that function out of this class and to the bottom of the module after tests since it is a helper method.